Use `RecentlyActive` feature to only list metrics older than 3 hours when necessary

nathan-beneke opened the pull request:
I have the problem that a lot of the dimensions on my CloudWatch metrics change frequently. Because the default behaviour of `listMetrics()` is to return every metric that matches your query and has been updated in the last 2 weeks, this causes the scrape time to be very high. With this option, we can limit the returned metrics to only those which have been updated in the last 3 hours. I also had to update the AWS SDK to allow this, as the version currently being used doesn't have this feature

cc @brian-brazil

brian-brazil commented:
If we're using this it shouldn't be configurable, we already know the range_seconds in use.

nathan-beneke commented:
So you would suggest to set this option if `range_seconds` < 3 hours? I guess that also makes sense

brian-brazil commented:
Yes.

nathan-beneke commented:
@brian-brazil I agree that's a lot simpler actually. Changed it to do that and updated the test to make sure that it's actually still possible to get metrics older than 3 hours with this setup

nathan-beneke commented:
I see this was also talked about in this open issue but never pursued further. With regard to the concern with S3 and similar metrics, I think it's okay because for those metrics the `range_seconds` has to be very high anyways https://github.com/prometheus/cloudwatch_exporter/issues/134#issuecomment-671185446_

brian-brazil commented:
Thanks!

guusvw commented:
Hey @brian-brazil 
What is your policy on building new releases? ;)

brian-brazil commented:
As-needed basically, I've added it to my todo list.

kjpianeta commented:
Hi @brian-brazil should we expect a new release with this in the next week(s), month(s)? Thanks.
